File: src/main/kotlin/com/huawei/cangjie/psi/CjImportList.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.psi.stubs.CangJiePlaceHolderStub;
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;

import java.util.List;



public class CjImportList extends CjElementImplStub<CangJiePlaceHolderStub<CjImportList>> {

    public CjImportList(@NotNull ASTNode node) {
        super(node);
    }

    public CjImportList(@NotNull CangJiePlaceHolderStub<CjImportList> stub) {
        super(stub, CjStubElementTypes.IMPORT_LIST);
    }

    @Override
    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, D data) {
        return visitor.visitImportList(this, data);
    }

    @NotNull
    public List<CjImportDirective> getImports() {
        return getStubOrPsiChildrenAsList(CjStubElementTypes.IMPORT_DIRECTIVE);
    }
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjImportList.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.psi.stubs.CangJiePlaceHolderStub
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode

class CjImportList : CjElementImplStub<CangJiePlaceHolderStub<CjImportList > > {
    constructor(node: ASTNode) : super(node)

    constructor(stub: CangJiePlaceHolderStub<CjImportList >) : super(stub, CjStubElementTypes.IMPORT_LIST)

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitImportList(this, data)
    }

    val imports: List<CjImportDirective>
        get() = getStubOrPsiChildrenAsList(CjStubElementTypes.IMPORT_DIRECTIVE)
}
